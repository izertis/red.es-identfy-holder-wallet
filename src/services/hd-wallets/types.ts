import { derivationPathFormats } from './constants'

type randomType = 'random'

export type settingType = {
	m?: undefined
	Z?: string
	R?: string | randomType
	SSSSS?: [string, string, string, string, string] | randomType
	W?: string | randomType
	M?: string
	T?: string
	N?: string
	B?: randomType
	BB?: [string, string] | randomType
}
export type validKeysType = keyof settingType

export type derivationPathFormat =
	| derivationPathFormats.MAIN_IDENTITY
	| derivationPathFormats.NETWORK_DID
	| derivationPathFormats.INTERACTING_DID
