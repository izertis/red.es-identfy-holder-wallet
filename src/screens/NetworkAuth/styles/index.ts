import defaultStyles from "./defaultStyles"
import proyectSyles from "./proyectStyles"

const NetworkAuthStyled = {
  ...defaultStyles,
  ...proyectSyles
}

export default NetworkAuthStyled
