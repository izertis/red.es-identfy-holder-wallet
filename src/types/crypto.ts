export type JWT = `ey${string}.ey${string}.${string}`