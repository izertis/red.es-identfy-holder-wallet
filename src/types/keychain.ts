import { ActionLog, CredentialData, DeferredCredential, PresentationData } from '../services/open-id/types'

export type KeyChainData = {
	encryptedPrivateKey?: string | ArrayBuffer
	did?: { lacchain?: string; alastria?: string; ebsi?: string }
	JWKKeys?: { privateKeyJWK: string; publicKeyJWK: string; privateKeyHex: string; publicKeyHex: string }
	signatureKeyPair?: { publicKey: string; privateKey: string }
	credentials?: CredentialData[]
	presentations?: PresentationData[]
	deferred_credentials?: DeferredCredential[]
	actionLogs?: ActionLog[]
}
