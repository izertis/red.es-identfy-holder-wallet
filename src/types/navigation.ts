import { MaterialTopTabScreenProps } from '@react-navigation/material-top-tabs'
import { NativeStackNavigationOptions } from '@react-navigation/native-stack'

export type navigationScreensConfigType = {
	name: string
	component: (props: any) => JSX.Element
	options:
		| NativeStackNavigationOptions
		| MaterialTopTabScreenProps<{}>
		| ((x: any) => NativeStackNavigationOptions | MaterialTopTabScreenProps<{}>)
}
