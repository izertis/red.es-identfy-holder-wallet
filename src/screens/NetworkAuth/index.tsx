import { useContext, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { KeyboardAvoidingView, View } from 'react-native'
import { Text } from 'react-native-paper'
import { RootStackScreenProps } from '../../../types'
import { SCREEN } from '../../constants/screens'
import { MessageContext } from '../../context/UserMessage.context'
import NetworkAuthI18nKeys from './i18n/keys'
import NetworkAuthStyled from './styles'
import { createDidLacchain } from '../../services/lacchain/createDid'
import { createDidEbsi } from '../../services/ebsi/createDid'
import { getKeychainDataObject } from '../../utils/keychain'
import { createDidAlastria } from '../../services/alastria/createDid'
import { StatusBar } from 'expo-status-bar'
import { ColorKeys, getThemeColor } from '../../constants/Colors'

interface Checkbox {
	entity: string,
	label: string
	subLabel: string
	checked: boolean
	action: () => Promise<void>
}

const NetworkAuth = ({ navigation }: RootStackScreenProps<'NetworkAuth'>): JSX.Element => {
	const { t } = useTranslation(SCREEN.NetworkAuth)

	const { showMessage } = useContext(MessageContext)
	const [dids, setDids] = useState<any>({})
	const [isLoading, setIsLoading] = useState<boolean>(false)

	useEffect(() => {
		const fetchData = async () => {
			const keychain = await getKeychainDataObject()
			setDids(keychain?.did || {})
		}
		fetchData()
	}, [])

	const checkboxes: Checkbox[] = [
		{
			entity: 'alastria',
			label: t(NetworkAuthI18nKeys.CHECKBOX_ALASTRIA),
			subLabel: t(NetworkAuthI18nKeys.CHECKBOX_ALASTRIA_SUB),
			checked: dids.alastria?.AlastriaT || false,
			action: async () => {
				return await createDidAlastria()
			},
		},
		{
			entity: 'lacchain',
			label: t(NetworkAuthI18nKeys.CHECKBOX_LACCHAIN),
			subLabel: t(NetworkAuthI18nKeys.CHECKBOX_LACCHAIN_SUB),
			checked: dids.lacchain || false,
			action: async () => {
				return await createDidLacchain()
			},
		},
		{
			entity: 'ebsi',
			label: t(NetworkAuthI18nKeys.CHECKBOX_EBSI),
			subLabel: t(NetworkAuthI18nKeys.CHECKBOX_EBSI_SUB),
			checked: dids.ebsi || false,
			action: async () => {
				return await createDidEbsi()
			},
		},
	]

	const [checkboxStates, setCheckboxStates] = useState<{ [key: string]: boolean }>(
		checkboxes.reduce((acc: any, curr) => {
			acc[curr.label] = curr.checked
			return acc
		}, {})
	)

	const handleCreateDID = async (): Promise<void> => {
		setIsLoading(true)
		let isSelected = false
		try {
			if (checkboxStates[t(NetworkAuthI18nKeys.CHECKBOX_ALASTRIA)]) {
				await checkboxes[0].action()
			}
			if (checkboxStates[t(NetworkAuthI18nKeys.CHECKBOX_LACCHAIN)]) {
				await checkboxes[1].action()
			}
			if (checkboxStates[t(NetworkAuthI18nKeys.CHECKBOX_EBSI)]) {
				await checkboxes[2].action()
			}

			for (const key in checkboxStates) {
				if (checkboxStates[key] === true) {
					isSelected = true
					break
				}
			}
			isSelected && showMessage({ content: t?.(NetworkAuthI18nKeys.SUCCESS), type: "success" })

		} catch (error) {
			showMessage({ content: t?.(NetworkAuthI18nKeys.ERROR), type: "error" })
		} finally {
			setIsLoading(false)
			navigation.reset({
				index: 0,
				routes: [{ name: 'Root' }],
			})
		}
	}

	const handleCheckboxChange = (label: string): void => {
		setCheckboxStates({
			...checkboxStates,
			[label]: !checkboxStates[label],
		})
	}
	const isCreateDidNeeded = Object.values(checkboxStates).some((checked) => !!checked)

	return (
		<KeyboardAvoidingView>
			<StatusBar backgroundColor='transparent' style='auto' />
			<NetworkAuthStyled.MainContainer>
				<NetworkAuthStyled.GoBackButtonContainer>
					{navigation.canGoBack() &&
						<NetworkAuthStyled.GoBackButton onPress={() => navigation.pop()}>
							{t(NetworkAuthI18nKeys.GO_BACK)}
						</NetworkAuthStyled.GoBackButton>
					}

				</NetworkAuthStyled.GoBackButtonContainer>

				<NetworkAuthStyled.Title testID='title'>{t(NetworkAuthI18nKeys.TITLE)}</NetworkAuthStyled.Title>

				<NetworkAuthStyled.Subtitle>{t(NetworkAuthI18nKeys.SUBTITLE)}</NetworkAuthStyled.Subtitle>

				{checkboxes.map((checkbox: Checkbox, index: number) => (
					<NetworkAuthStyled.CheckBoxContainer key={index}>
						<NetworkAuthStyled.CheckBox
							checked={checkboxStates[checkbox.label] || dids[checkbox.entity]}
							disabled={!!dids[checkbox.entity] || isLoading}
							onPress={() => handleCheckboxChange(checkbox.label)}
						/>
						<View>
							<Text style={{ fontWeight: 'bold' }}>{checkbox.label}</Text>
							<Text>{checkbox.subLabel}</Text>
						</View>
					</NetworkAuthStyled.CheckBoxContainer>
				))}

				<NetworkAuthStyled.Subtitle>{t(NetworkAuthI18nKeys.FOOTER_TEXT)}</NetworkAuthStyled.Subtitle>

				<NetworkAuthStyled.ContainerBottom>
					{isLoading ? <NetworkAuthStyled.SplashActivityIndicator color={getThemeColor(ColorKeys.text)} size={35} /> : null}
					<NetworkAuthStyled.Button
						testID='create-did-button'
						disabled={isLoading}
						onPress={() => handleCreateDID()}
					>
						{isLoading
							? t(NetworkAuthI18nKeys.LOADING)
							: isCreateDidNeeded
								? t(NetworkAuthI18nKeys.BUTTON)
								: t(NetworkAuthI18nKeys.CONTINUE)
						}
					</NetworkAuthStyled.Button>
				</NetworkAuthStyled.ContainerBottom>
			</NetworkAuthStyled.MainContainer>
		</KeyboardAvoidingView>
	)
}

export default NetworkAuth
