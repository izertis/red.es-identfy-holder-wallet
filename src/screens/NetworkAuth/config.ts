import i18next from '../../../i18n.config'
import { SCREEN } from "../../constants/screens"
import { navigationScreensConfigType } from "../../types/navigation"
import es from "./i18n/es"
import NetworkAuth from '.'

const ScreenName = SCREEN.NetworkAuth

i18next.addResourceBundle("es", ScreenName, es)

const NetworkAuthConfig: navigationScreensConfigType = {
  name: ScreenName,
  component: NetworkAuth,
  options: { headerShown: false },
}

export default NetworkAuthConfig
