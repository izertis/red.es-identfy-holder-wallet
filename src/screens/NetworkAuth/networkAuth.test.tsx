import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react-native'
import NetworkAuth from '.'
import { NativeStackNavigationProp } from '@react-navigation/native-stack'
import { RootStackParamList } from '../../../types'
import { RouteProp } from '@react-navigation/native'

describe('NetworkAuth', () => {
  type ScreenNavigationProp = NativeStackNavigationProp<
    RootStackParamList,
    'NetworkAuth'
  >
  type NetworkAuthRouteProp = RouteProp<RootStackParamList, 'NetworkAuth'>

  const mockedNavigation: ScreenNavigationProp | any = {
    navigate: jest.fn(),
    route: jest.fn(),
  }

  const mockedRoute: NetworkAuthRouteProp = {
    key: 'mocked-key',
    name: 'NetworkAuth',
  }

  it('should render NetworkAuth screen', async () => {
    const result = render(
      <NetworkAuth navigation={mockedNavigation} route={mockedRoute} />
    )

    await waitFor(() => {
      expect(result.toJSON()).toMatchSnapshot()
    })
  })

  it('renders title correctly', async () => {
    const { getByTestId } = render(
      <NetworkAuth navigation={mockedNavigation} route={mockedRoute} />
    )

    await waitFor(() => {
      expect(getByTestId('title')).toBeTruthy()
    })
  })

  it('renders checkboxes and button', async() => {
    const { getAllByRole, getByTestId } = render(
      <NetworkAuth navigation={mockedNavigation} route={mockedRoute} />
    )
    const checkboxes = getAllByRole('checkbox')
    const button = getByTestId('create-did-button')

    await waitFor(() => {
      expect(checkboxes.length).toBe(3)
      expect(button).toBeDefined()
    })
  })
})
