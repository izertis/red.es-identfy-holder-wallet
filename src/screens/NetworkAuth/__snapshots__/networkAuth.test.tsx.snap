// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkAuth should render NetworkAuth screen 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Array [
        Object {
          "height": "100%",
          "marginLeft": 25,
          "marginRight": 25,
          "marginTop": 15,
          "paddingBottom": 35,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "marginBottom": "6%",
          "marginLeft": "-1%",
          "marginTop": "5%",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": "22%",
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 20,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginLeft": 12,
                    "marginRight": -4,
                  },
                  Object {
                    "marginLeft": 16,
                    "marginRight": -16,
                  },
                  Object {
                    "marginLeft": 12,
                    "marginRight": -8,
                  },
                ]
              }
              testID="button-icon-container"
            >
              <Text />
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 16,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#eb5b5b",
                      },
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              GO_BACK
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Text
      bold={true}
      className="sc-bcXHqe sc-hLBbgP sc-gKPRtg kFXtiY RwpaV"
      fontFamily="Montserrat"
      style={
        Array [
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          Object {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
      testID="title"
    >
      TITLE
    </Text>
    <Text
      className="sc-bcXHqe sc-dkrFOg sc-iBYQkv laZgtI bNRHGr htBlcA"
      style={
        Array [
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          Object {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
    >
      SUBTITLE
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingTop": "5%",
          },
        ]
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        color="#363d58"
        containerStyle={
          Object {
            "marginHorizontal": 5,
            "padding": 4,
          }
        }
        focusable={true}
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [
              [Function],
              [Function],
            ],
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
	font-size: 14px;
	font-weight: normal;
	font-family: ",
                "Muli",
                ";
	border-width: 1px;
",
                "",
              ],
            },
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "padding": 6,
              "width": 36,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text />
          <View
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "borderColor": "rgba(73, 69, 79, 1)",
                  "borderWidth": 0,
                  "height": 14,
                  "width": 14,
                }
              }
            />
          </View>
        </View>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          CHECKBOX_ALASTRIA
        </Text>
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              Object {
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          CHECKBOX_ALASTRIA_SUB
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingTop": "5%",
          },
        ]
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        color="#363d58"
        containerStyle={
          Object {
            "marginHorizontal": 5,
            "padding": 4,
          }
        }
        focusable={true}
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [
              [Function],
              [Function],
            ],
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
	font-size: 14px;
	font-weight: normal;
	font-family: ",
                "Muli",
                ";
	border-width: 1px;
",
                "",
              ],
            },
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "padding": 6,
              "width": 36,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text />
          <View
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "borderColor": "rgba(73, 69, 79, 1)",
                  "borderWidth": 0,
                  "height": 14,
                  "width": 14,
                }
              }
            />
          </View>
        </View>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          CHECKBOX_LACCHAIN
        </Text>
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              Object {
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          CHECKBOX_LACCHAIN_SUB
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingTop": "5%",
          },
        ]
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        color="#363d58"
        containerStyle={
          Object {
            "marginHorizontal": 5,
            "padding": 4,
          }
        }
        focusable={true}
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [
              [Function],
              [Function],
            ],
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
	font-size: 14px;
	font-weight: normal;
	font-family: ",
                "Muli",
                ";
	border-width: 1px;
",
                "",
              ],
            },
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "padding": 6,
              "width": 36,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text />
          <View
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "borderColor": "rgba(73, 69, 79, 1)",
                  "borderWidth": 0,
                  "height": 14,
                  "width": 14,
                }
              }
            />
          </View>
        </View>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          CHECKBOX_EBSI
        </Text>
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              Object {
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          CHECKBOX_EBSI_SUB
        </Text>
      </View>
    </View>
    <Text
      className="sc-bcXHqe sc-dkrFOg sc-iBYQkv laZgtI bNRHGr htBlcA"
      style={
        Array [
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          Object {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
    >
      FOOTER_TEXT
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "flex-end",
            "marginBottom": 40,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 20,
            "width": "100%",
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="create-did-button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flex": undefined,
              "minWidth": 64,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="create-did-button-container"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 20,
                },
              ]
            }
            testID="create-did-button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "#eb5b5b",
                    "borderRadius": 15,
                    "fontFamily": "Muli",
                    "height": 39,
                    "width": 224,
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      Array [
                        Object {
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        false,
                        Object {
                          "marginHorizontal": 12,
                        },
                        undefined,
                        false,
                        Object {
                          "color": "rgba(103, 80, 164, 1)",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        Object {
                          "color": "#000",
                          "fontFamily": "Muli",
                          "letterSpacing": 0.57,
                        },
                      ],
                    ],
                  ]
                }
                testID="create-did-button-text"
              >
                CONTINUE
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
